lib/transports/websocket.js

```diff
@@ -87,8 +88,22 @@ WebSocket.prototype.send = function (packets) {
   packets.forEach(function(packet) {
     parser.encodePacket(packet, self.supportsBinary, function(data) {
       debug('writing "%s"', data);
+
+      // always creates a new object since ws modifies it
```

Comment:
It's not a bug of `ws`. Obviously, it's not good, but `ws` just works like that. 

https://github.com/nkzawa/ws/blob/master/lib/WebSocket.js#L230